services/ingestion/src/pdf_ingestion.py

```diff
@@ -9,31 +9,40 @@
 import click
 import tempfile
 import time
-import subprocess
 import os
 import glob
+import subprocess
+import io
+from preprocess import resize_png
 from PIL import Image
 from typing import Mapping, TypeVar, Callable
 from pdf_extractor import parse_pdf
 from pymongo import MongoClient
+from bson.objectid import ObjectId
 import json
 
 
 T = TypeVar('T')
 
-def run_pdf_ingestion(pdf_dir: str, db_insert_fn: Callable[[Mapping[T, T]], None], subprocess_fn: Callable[[str, str], None]) -> Mapping[T, T]:
+def run_pdf_ingestion(pdf_dir: str, db_insert_fn: Callable[[Mapping[T, T]], None], db_insert_pages_fn: Callable[[Mapping[T,T]], None], subprocess_fn: Callable[[str, str], None]) -> Mapping[T, T]:
     """
     Entry point for ingesting PDF documents
     """
+
     logging.info('Running ingestion')
     start_time = time.time()
     pdfs = []
     for pdf_path in glob.glob(os.path.join(pdf_dir, '*.pdf')):
         with tempfile.TemporaryDirectory() as img_tmp:
-            subprocess_fn(pdf_path, img_tmp)
             pdf_obj = {}
-            pdf_obj = load_page_data(img_tmp, pdf_obj)
             pdf_obj = load_pdf_metadata(pdf_path, pdf_obj)
+            pdf_obj["_id"] = ObjectId() # Want to know the _id _before_ insertion so we can tag pages in their collection
+            pdf_path = os.path.abspath("%s/%s" % (pdf_dir, pdf_obj["pdf_name"]))
+            # TODO: get real pdf_path as pdf_dir + pdf_name (from doc)
```

Comment:
Could you expand on this TODO?

Reply:
Looks like I just forgot to rm it after I added the line above. My bad!